SyntaxKit/AttributedParser.swift

```diff
@@ -43,6 +43,16 @@ open class AttributedParser: Parser {
         }
     }
 
+    open func addAttributes(to mutableString: NSMutableAttributedString) {
+        mutableString.beginEditing()
```

Comment:
Does it make a difference if you remove all the the foreground attributes first?